modules/ref_fb_module/src/fft_fb_impl.cpp

```diff
@@ -0,0 +1,262 @@
+            inputSampleType != SampleType::Int16 &&
+            inputSampleType != SampleType::Int32 &&
+            inputSampleType != SampleType::Int64 &&
+            inputSampleType != SampleType::UInt8 &&
+            inputSampleType != SampleType::UInt16 &&
+            inputSampleType != SampleType::UInt32 &&
+            inputSampleType != SampleType::UInt64)
+            throw std::runtime_error("Invalid sample type");
+
+        if (inputDomainDataDescriptor.getSampleType() != SampleType::Int64 && inputDomainDataDescriptor.getSampleType() != SampleType::UInt64)
+        {
+            throw std::runtime_error("Incompatible domain data sample type");
+        }
+
+        const auto domainUnit = inputDomainDataDescriptor.getUnit();
+        if (domainUnit.getSymbol() != "s" && domainUnit.getSymbol() != "seconds")
+        {
+            throw std::runtime_error("Domain unit expected in seconds");
+        }
+
+        const auto domainRule = inputDomainDataDescriptor.getRule();
+        if (inputDomainDataDescriptor.getRule().getType() != DataRuleType::Linear)
+        {
+            throw std::runtime_error("FFT: Domain rule must be linear");
+        }
+
+        linearReader = BlockReaderFromExisting(linearReader, blockSize, SampleType::Float32, SampleType::UInt64);
+
+        auto dimensions = List<IDimension>();
+        dimensions.pushBack(Dimension(DimensionRuleBuilder()
+                                      .setType(DimensionRuleType::Linear)
+                                      .addParameter("delta", 1)
+                                      .addParameter("start", 1)
+                                      .addParameter("size", blockSize / 2)
+                                      .build()));
+
+        const auto inputRange = inputDataDescriptor.getValueRange();
+        outputDataDescriptor = DataDescriptorBuilder()
+                                   .setSampleType(SampleType::Float64)
+                                   .setDimensions(dimensions)
+                                   .setUnit(inputDataDescriptor.getUnit())
+                                   .setValueRange(inputRange.assigned() ? Range(0, inputRange.getHighValue()) : Range(0, 10))
+                                   .build();
+
+        outputSignal.setDescriptor(outputDataDescriptor);
+
+        outputDomainDataDescriptor =
+            DataDescriptorBuilderCopy(inputDomainDataDescriptor).setRule(ExplicitDataRule()).setSampleType(SampleType::UInt64).build();
+        outputDomainSignal.setDescriptor(outputDomainDataDescriptor);
+            
+        inputData = std::make_unique<float[]>(maxBlockReadCount * blockSize);
+        inputDomainData = std::make_unique<uint64_t[]>(maxBlockReadCount * blockSize);
+        
+        kiss_fft_free(cfg);
+        cfg = kiss_fft_alloc(static_cast<int>(blockSize), 0, nullptr, nullptr);
+        fftIn = std::make_unique<kiss_fft_cpx[]>(blockSize);
+        fftOut = std::make_unique<kiss_fft_cpx[]>(blockSize);
```

Comment:
I would just use std::vector with .resize()/.data() instead of unique_ptrs to naked arrays for the buffers.